### PR TITLE
EDTR: DatesList cleanup.

### PR DESCRIPTION
**Pull request by nicolad**
## Problem this Pull Request solves
Improves code based on this: https://github.com/eventespresso/event-espresso-core/issues/1918

## Checklist

* [ ] I have read the documentation relating to systems affected by this pull request, see https://github.com/eventespresso/event-espresso-core/tree/master/docs
* [ ] User input is adequately validated and sanitized
* [ ] all publicly displayed strings are internationalized (usually using `esc_html__()`, see https://codex.wordpress.org/I18n_for_WordPress_Developers)
* [ ] My code is tested.
* [ ] My code follows the Event Espresso code style.
* [ ] My code has proper inline documentation.
* [ ] My code accounts for when the site is in Maintenance Mode (MM2 especially disallows usage of models)

### CODE REVIEW

File: assets/prototype/domain/eventEditor/containers/queries/useInitQueries.js [hunk 1]
```diff
@@ -32,7 +33,7 @@ const useInitQueries = ({ eventId }) => {
 		toaster.error(datetimeError);
 	}
 
-	const datetimes = get(datetimesData, ['datetimes', 'nodes']);
+	const datetimes = pathOr([], ['datetimes', 'nodes'], datetimesData);
```
Comment (nicolad) [inline assets/prototype/domain/eventEditor/containers/queries/useInitQueries.js]: The default value will reduce the boilerplate part:
`Array.isArray(datetimes) &&`

File: assets/prototype/domain/eventEditor/containers/queries/useInitQueries.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import get from 'lodash/get';
```
Comment (tn3rb) [inline assets/prototype/domain/eventEditor/containers/queries/useInitQueries.js]: should remove this import if it is no longer being used

File: assets/prototype/domain/eventEditor/datesList/DatesList.js [hunk 3]
```diff
@@ -1,18 +1,11 @@
-import { Callout, H3, H6, NonIdealState, Spinner } from '@blueprintjs/core/lib/esm';
+import { H3 } from '@blueprintjs/core/lib/esm';
+import LoadingIndicator from '../../shared/LoadingIndicator';
+import ErrorIndicator from '../../shared/ErrorIndicator';
+import EmptyState from '../../shared/EmptyState';
```
Comment (tn3rb) [inline assets/prototype/domain/eventEditor/datesList/DatesList.js]: 👍

File: assets/prototype/domain/eventEditor/containers/queries/useInitQueries.js (hunk 2 above)
Comment (nicolad) [reply assets/prototype/domain/eventEditor/containers/queries/useInitQueries.js]: Sure thing, thanks, updated.

**Pull request merged**